packages/pwa-kit-runtime/src/utils/logger.js

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+const LOG_LEVELS = ['debug', 'info', 'warn', 'error']
+const DEFAULT_LOG_LEVEL = 'info'
+const DEFAULT_LOG_FORMAT = 'JSON'
+
+const isServerSide = typeof window === 'undefined'
+
+const LOG_LEVEL = isServerSide
+    ? process.env.PWAKIT_LOG_LEVEL || DEFAULT_LOG_LEVEL
+    : DEFAULT_LOG_LEVEL
+
+const LOG_FORMAT = isServerSide
+    ? process.env.PWAKIT_LOG_FORMAT || DEFAULT_LOG_FORMAT
+    : DEFAULT_LOG_FORMAT
+
+/**
+ * The PWAKITLogger provides structured logging with different log levels.
+ *
+ */
+export class PWAKITLogger {
+    /**
+     * Creates an instance of PWAKITLogger.
+     * @param {string} logLevel - The log level to set for the logger.
+     * @param {string} [format=DEFAULT_LOG_FORMAT] - The format in which to print log messages. Can be 'JSON' or 'TEXT'.
+     * @param {string} packageName - The name of the package where the logger is used.
+     */
+    constructor(logLevel = LOG_LEVEL, format = DEFAULT_LOG_FORMAT, packageName = '') {
```

Comment:
If you had 2 instances of the logger class side by side, what would be the primary means of discerning the 2? You'd probably say that the "packageName" would be difference. And I would agree with that. So in essence the packageName is like an "id" prop. And typically the id would be the first argument in a constructor. 

what do you think?

Reply:
I might also lean towards making the argument an object.. this makde save you in the future if you were to want to come back and add an argument to this constructors property list.. you might ends up with something like (level, format, id, priority) where things get a little mixed up

Reply:
It makes sense to use an object for the constructor argument. I made the change.